### PR TITLE
ft-sockets

### PR DESCRIPTION
**Pull request by JPaul23**
### What does this PR do?
Add functionality for listening, emitting and broadcasting events between clients(Web APP) and server using `Socket.io` that provides bidirectional and low-latency communication for every platform.
### Why this feature?
To enable sending events from driver dashboard to user map side, to display real-time status of bus across map.
### How was the feature implemented?
This feature was implemented as a sequel of existing bus-simulation, where controls can now send events that can be heard across the application.
### How to test this PR
1. Clone the app both backend and frontend. `git clone {REPOSITORY_URL}` change the {REPOSITY_URL} with the actual link.
2. Run npm install both backend and frontend. `npm install`, this will install all dependencies
3. Edit the **env** to include the *REACT_APP_URL* with the frontend URL.
4. Start backend and frontend servers.
5. Select the routes from /user-map.
6. Start the bus on /simulator.
7. You can stop the bus by pressing Stop button.

Visit [socket.io doc](https://socket.io/docs/v4/)

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -96,10 +97,33 @@ app.use(unassignDriver)
...
+const frontedUrl = process.env.REACT_APP_URL
+
+const io = Server(serverApp, {
+  cors: {
+    origin: frontedUrl,
+    methods: ['GET', 'POST']
+  }
+})
+
+const clients = []
```
Comment (Anguandia) [inline src/app.js]: I think we do not really need this for now

File: src/app.js [hunk 2]
```diff
@@ -96,10 +97,33 @@ app.use(unassignDriver)
...
+
+io.on('connection', (client) => {
+  // logging clients
+  client.push(client.id)
+  console.log(`==> User with ${client.id} connected || Total Clients: ${clients.length}`)
+
```
Comment (Anguandia) [inline src/app.js]: And theses

**Review by Anguandia — CHANGES_REQUESTED** [review]
Good work @JPaul23 , please resolve the conflicts with develop, then we'll be good to go

**Review by Anguandia — APPROVED** [review]

**Review by SimonImanigirimpuhwe — APPROVED** [review]
LGTM

**Pull request merged**